src/helics/network/tcp/TcpHelperClasses.cpp

```diff
@@ -257,13 +257,27 @@ namespace tcp {
                 return 0;
             }
         }
-        auto sz = socket_.send(asio::buffer(buffer, dataLength));
-        assert(sz == dataLength);
-        return sz;
+	
+	size_t	sz,sent_size;
+	int p=0,count=0;
+	sent_size=dataLength;
+        while (count++  < 5 && (sz = socket_.send(asio::buffer((void *)((char *)buffer+p), sent_size))) != sent_size) {
+	   sent_size-=sz; p+=sz;
+	//   std::cerr << "DEBUG partial buffer sent" << std::endl;
+	}
+	if(count < 5 ) return dataLength;
```

Comment:
This should be count<=5 if I am not mistaken?   and any particular reason for 5 being chosen here?

Reply:
From what I saw while debugging, one  extra resend works in most cases, sometimes it requires two. The max number of sends it set to  4 (sorry, not 5) to eliminate some false positives,  so it should be sufficient for now I believe. If it would require more than 4, then most likely it will be going for a very large number of resends. It may be beneficial to add a very short sleep between resends, but it may negatively affect the performance.